### PR TITLE
KMS: Updated for new client library

### PR DESCRIPTION
**Pull request by DanSanche**
Updated all samples to use the new libraries

tracking bug: b/120216201

### CODE REVIEW

File: kms/src/main/java/com/example/Snippets.java [hunk 1]
```diff
@@ -16,57 +16,43 @@
...
-    // Check for this case, and inject the scope if required.
-    if (credential.createScopedRequired()) {
-      credential = credential.createScoped(CloudKMSScopes.all());
-    }
-
-    return new CloudKMS.Builder(transport, jsonFactory, credential)
-        .setApplicationName("CloudKMS snippets")
-        .build();
+  public static KeyManagementServiceClient createClient() throws IOException {
+    // GAP: missing ability to set application name
```
Comment (lesv) [inline kms/src/main/java/com/example/Snippets.java]: should more be said here?

File: kms/src/main/java/com/example/Quickstart.java [hunk 2]
```diff
@@ -19,71 +19,39 @@
...
 
   public static void main(String... args) throws Exception {
     String projectId = args[0];
     // The location of the Key Rings
     String location = args[1];
 
-    // Create the Cloud KMS client.
-    CloudKMS kms = createAuthorizedClient();
+    // Create the KeyManagementServiceClient using try-with-resources to manage client cleanup.
+    // GAP: missing ability to set application name
```
Comment (lesv) [inline kms/src/main/java/com/example/Quickstart.java]: should more be said here?

File: kms/src/main/java/com/example/Snippets.java [hunk 3]
```diff
@@ -437,38 +399,42 @@ public static Policy removeMemberFromCryptoKeyPolicy(
...
+
+      // The resource name of the cryptoKey version
+      String keyName = CryptoKeyName.format(projectId, locationId, keyRingId, cryptoKeyId);
+
+      // Get the current IAM policy
+      Policy iamPolicy = client.getIamPolicy(keyName);
+
+      // Create a bindings list that filters out the provided member
+      List<Binding> newBindings = new ArrayList<>();
+      for (Binding b : iamPolicy.getBindingsList()) {
```
Comment (lesv) [inline kms/src/main/java/com/example/Snippets.java]: Go prefer's short names, Java likes a bit longer than 'b'.

File: kms/src/main/java/com/example/Snippets.java [hunk 4]
```diff
@@ -480,136 +446,135 @@ public static Policy removeMemberFromCryptoKeyPolicy(
...
+
+      // The resource name of the cryptoKey version
+      String keyRingName = KeyRingName.format(projectId, locationId, keyRingId);
+
+      // Get the current IAM policy
+      Policy iamPolicy = client.getIamPolicy(keyRingName);
+
+      // Create a bindings list that filters out the provided member
+      List<Binding> newBindings = new ArrayList<>();
+      for (Binding b : iamPolicy.getBindingsList()) {
```
Comment (lesv) [inline kms/src/main/java/com/example/Snippets.java]: 'binding' ?

File: kms/src/test/java/com/example/AsymmetricIT.java [hunk 5]
```diff
@@ -55,65 +48,63 @@
...
       return true;
+    } catch (AlreadyExistsException e) {
+      return false;
     }
   }
 
   @BeforeClass
   public static void setUpClass() throws Exception {
-    client = Asymmetric.createAuthorizedClient();
+    // TODO: set application name
```
Comment (lesv) [inline kms/src/test/java/com/example/AsymmetricIT.java]: want to mention the github issue here?

File: kms/src/test/java/com/example/CryptFileIT.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 Google Inc.
```
Comment (lesv) [inline kms/src/test/java/com/example/CryptFileIT.java]: Google LLC ?

File: kms/src/test/java/com/example/CryptFileIT.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+  /**
+   * Destroys all the key versions created during this test run.
+   */
+  @AfterClass
+  public static void tearDownClass() throws Exception {
+    List<CryptoKeyVersion> versions = 
+        Snippets.listCryptoKeyVersions(PROJECT_ID, LOCATION_ID, KEY_RING_ID, CRYPTO_KEY_ID);
+
+    for (CryptoKeyVersion v : versions) {
```
Comment (lesv) [inline kms/src/test/java/com/example/CryptFileIT.java]: v is a bit terse

File: kms/src/test/java/com/example/SnippetsIT.java [hunk 8]
```diff
@@ -41,312 +43,241 @@
...
-    String[] lines = stdout.split("\n");
-    Pattern keyVersion = Pattern.compile(
-        ".*cryptoKeyVersions/(\\d+)\",\"state\":\"(EN|DIS)ABLED\".*",
-        Pattern.DOTALL | Pattern.MULTILINE);
-
-    for (String line : lines) {
-      Matcher matcher = keyVersion.matcher(line);
-      if (!matcher.matches()) {
-        continue;
+    for (CryptoKeyVersion v : versions) {
```
Comment (lesv) [inline kms/src/test/java/com/example/SnippetsIT.java]: v is terse

File: kms/src/main/java/com/example/Asymmetric.java [hunk 9]
```diff
@@ -16,96 +16,121 @@
...
+  public static CryptoKey createAsymmetricKey(String projectId, String locationId, String keyRingId,
+      String cryptoKeyId)
+      throws IOException {
+
+    // Create the Cloud KMS client.
+    try (KeyManagementServiceClient client = KeyManagementServiceClient.create()) {
+      // The resource name of the location associated with the KeyRing.
+      String parent = KeyRingName.format(projectId, locationId, keyRingId);
+
+      // Choose a purpose (ASYMMETRIC_DECRYPT or ASYMMETRIC_SIGN).
```
[the remark this answers is not in the document]
Comment (DanSanche) [reply kms/src/main/java/com/example/Asymmetric.java]: I wanted to make it clear to readers how to modify this function for all combinations of asymmetric sign/decrypt/rsa/ec, but in hindsight it's probably self-documenting enough as it is, so I'll remove some of the extra comments

**Pull request merged**